base/multidimensional.jl

```diff
@@ -247,18 +247,25 @@ end
 end
 
 # checksize ensures the output array A is the correct size for the given indices
-@noinline throw_checksize_error(arr, dim, idx) = throw(DimensionMismatch("index $d selects $(length(I[d])) elements, but size(A, $d) = $(size(A,d))"))
+@noinline throw_checksize_error(A, dim, idx) = throw(DimensionMismatch("index $dim selects $(length(idx)) elements, but size(A, $dim) = $(size(A,dim))"))
+@noinline throw_checksize_error(A, dim, idx::AbstractArray{Bool}) = throw(DimensionMismatch("index $dim selects $(sum(idx)) elements, but size(A, $dim) = $(size(A,dim))"))
```

Comment:
This is checking the size of the output array… it's essentially ensuring that `similar` returned an object of the correct size.  It's of dubious value (looks like nobody hit the error since I had also introduced a bug in the error message), but it's cheap.  IIRC, it's what the builtin array types did before #10525, but my memory there is fuzzy.